client/gutenberg/extensions/gif/edit.js

```diff
@@ -11,6 +11,7 @@ import { debounce } from 'lodash';
 import { icon } from './';
 
 const GIPHY_API_KEY = 't1PkR1Vq0mzHueIFBvZSZErgFs9NBmYW';
+const SEARCH_INPUT_DEBOUNCE = 300; // Time before searching user input in ms
```

Comment:
300 ms is still too short for me to finish typing my word. I only managed to type 2 letters.

Reply:
I've made a significant increase to 450ms. How's that feel?

Reply:
Part of the reason early search is jarring is that the block may be resized which may move the input, and the input is hidden very quickly.

In fact, you should be able to just keep on typing without interruption while search occurs in the background, but the experience interrupts you. I think it's important to smooth out the feel independent of the timing, although we might leave that for a future iteration.

Reply:
It's much better, thank you @sirreal!

I didn't realise you could still type and the early "jumping" surprised me and was unexpected as I thought it would start searching once I fully stopped typing.

Reply:
> I didn't realise you could still type and the early "jumping" surprised me and was unexpected as I thought it would start searching once I fully stopped typing.

My point exactly, it would be nice to smooth that out.

Reply:
#30408 may improve the feel a bit, see what you think

Reply:
Yep, this is smoother. 💯